client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -250,6 +250,8 @@ async fn sign_all_messages_and_send<T: Signers + ?Sized>(
                 ),
             );
         }
+
+        tokio::time::sleep(Duration::from_millis(500)).await;
```

Comment:
looks unrelated to the primary change?

Reply:
`program::tests::test_cli_keypair_file` are failing because of tokio scheduling. This is because in  tests tokio will be stuck in loop on this  task, and cannot schedule other tokio tasks.

Reply:
:thinking: 

sounds broken :detective:

Reply:
Not really, in real scenario tokio will wait for rpc_client to finish i/o request from jsonrpc client. So tokio is able to schedule tasks correctly. In test this rpc_client is mocked and so there is no real i/o that is why the task looped and finished all its tries. So this was not a real problem, it is just because the rpc_client in test is mocked and not really doing any i/o.

Reply:
This change will also reduce load on the RPC, it will wait at least 500ms before again sending the request `get_signature_status` to the RPC.

Reply:
Ok update mock rpc client to represent latency between rpc server and rpc client.

Reply:
Fixed and recommited.

Reply:
After some more analysis : https://github.com/solana-labs/solana/issues/32792 bug. Rpc client is no longer used incase of failure to send over TPU client. Currently, there is no way around a small wait to handle scheduling correctly.
I can just wait for 1ms to help tokio schedule the tasks correctly.